### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by ronzle** [20c/vaping#52]
Hi,

first of all, this project is great !!

2 Questions 

Is there a way to make the data persistent ?

I played a bit with a multilocation setup.
I set the on a client connect: tcp://remote-gunicorn-server:6021
Changed the bind address on the server, but no luck 

i was able to see the traffic in tcpdump, but there were no graphs generated.
Is this planned or already possible ?

Server
=====

data:
  - type: fping
    handlers:
      - type: index
        index: host
      - type: store
        container: list
        limit: 500

apps:
  graphsrv:
    tmpl_engine: jinja2
    enabled: true

    # here is where we define host config
    groups:
      latency:
        public_dns:
          8.8.8.8:
            name: Google DNS
            color: red
          4.2.2.1:
            name: Level(3)
            color: blue
          208.67.222.222:
            name: OpenDNS
            color: orange

    graphs:
      multitarget:
        id_field: host
        type: multitarget
        plot_y: avg
        format_y: ms

      smokestack:
        id_field: host
        type: smokestack
        plot_y: avg

plugins:

  # zero mq probe plugin (latency name is important, so it can be
  # routed properly to the similarly named group)
  - name: test
    type: zeromq_probe
    data: fping
    interval: 1.0
    bind: tcp://127.0.0.1:6021
    async: thread

  - name: http
    type: flask
    host: 0.0.0.0
    port: 7021
    debug: true
    # set this to gunicorn or uwsgi depending on what you want to run with
    server: gunicorn
    async: gevent
    routes:
      /targets : graphsrv->targets
      /graph_data :
        methods:
          - POST
          - GET
        target: graphsrv->graph_data
      /graph : graphsrv->graph_view
      /overview_read_file : graphsrv->overview_read_file
      /: graphsrv->overview_view

logging:
  version: 1
  formatters:
    simple:
      format: '%(asctime)s - %(name)s - %(levelname)s: %(message)s'
  handlers:
    console:
      class: logging.StreamHandler
      level: DEBUG
      formatter: simple
      stream: ext://sys.stdout
  loggers:
    vodka:
      level: DEBUG
      handlers:
        - console

Client 
=====
probes:
  - name: latency
    type: std_fping
    output:
      - zmq_vodka

    public_dns:
      hosts:
        - host: 1.1.1.1
        - host: 8.8.8.8

plugins:
  - name: std_fping
    type: fping
    count: 10
    interval: 3s

  - name: zmq_vodka
    type: zeromq
    connect: tcp://remote-gunicorn-server:6021

Thanks
Ronny

**Comment by vegu** [comment]
Hi Ronny,

1) there is currently no way to make data persistent in the web front end. you could how ever output it into whisper or rrd and save it there. Persistence in the web ui is planned, but no eta at this point.

2) all the examples are currently implemented and working - the only issue i can spot in your config pastes is that the client one should be `bind` as well. This is an odd config naming bug in the vodka zmq plugin and will be addressed in the future.

Let me know if that doesn't fix it and ill take a closer look at your config :)

**Comment by ronzle** [comment]
Hi 

1. ok Fine ;) i‘ll try the whisper 

2. did I mentioned that server and client are two different machines?
I had the ‚bind‘ set in the client config ... but I didn’t see the traffic on the master .

Thanks for helping I really appreciate that 

/ronny

**Comment by vegu** [comment]
Yeah i got the part of two different servers, with have that exact same setup running at https://vaping.20c.com where latency.dev is coming in from another server.

I will take a look at your configs in a bit

**Comment by ronzle** [comment]
To make clear the setup I would like to achieve is to monitor a group of hosts from different regions and push the data to one master host, which creates the graphs 

Thanks 
Ronny

**Comment by ronzle** [comment]
Sorry I saw your answer to late ;) 
If you want i can send you the configs ;) ?

**Comment by vegu** [comment]
sure send em to vegu @ 20c.com please
